test/classes/delete-free/borrowed/reassignBorrowed.chpl

```diff
@@ -0,0 +1,17 @@
+class C {
+  proc f() {
+    writeln("Called C.f()");
+  }
+}
+
+class D: C {
+  override proc f() {
+    writeln("Called D.f()");
+  }
+}
+
+var bc: borrowed C? = (new owned C()).borrow();
+bc!.f(); // OK. bc initializer will live to end of scope
+
+bc = (new owned D()).borrow();
+bc!.f(); // Not OK. bc is nil. It was freed at the end of the previous statement
```

Comment:
Should this test have a .noexec?

Reply:
> Should this test have a .noexec?

I don't think it's necessary but it can if you'd prefer that. Any output other than the expected error message is a failure so it won't be noisy if allowed to execute.

Reply:
Right, but when the output changes, we will get email about the new failure mode. I think?

Reply:
I don't think so. If I gave it a .bad file and it failed in a new way that didn't match the .bad file it would cause an email, but I didn't add a .bad file to avoid that. I'm pretty sure changing output won't cause new emails for this test unless the incorrect output miraculously matches the .good file.